### DISCUSSION TITLE
Extra space at end of table

### DISCUSSION THREAD

**Discussion by scottb5612** [Textualize/rich#3591] [General]
When I print a table   I get a large amount of
extra  blanks at the end of each row. I've tried outputting to a
file, and also to the terminal (and redirecting output to a file)
 How can I prevent this artifact?

I've tired various settings - show_edge, padding, collapse_padding, pad_edge,
but to no avail. Keeping the enclosing box doesn't avoid the padding either.

-- Code below ---

from rich import print
from rich.console import Console
from rich.logging import RichHandler
from rich.table import Table

console_sh = Console(file=open('Z.sh','w'))
RichHandler(console=console_sh)

tab = Table(show_lines=False, box=None)
""" An enclosing box doesn't avoid the problem either
 tab = Table(show_lines=False)
"""

tab.add_column("GeoHash", style=None, no_wrap=True, justify="right")
tab.add_column("Count", style=None, no_wrap=True, justify="right")

tab.add_row( "9q8zpu", "1")
tab.add_row( "u6sc6", "2")

console_sh.print(tab, justify="left")

**Comment by TomJGooding** [comment]
I'm not sure what you mean by "extra space", could you maybe show a screenshot?

Running your example (with just the regular `Console`) looks as expected to me.

**Comment by scottb5612** [comment]
I've attached an output file and a screen shot of a hex and character dump
Not possible to see the problem unless you can access the file itself  or a representation
that clearly shows the spaces.

There should be no   blanks (hex 20) following the vertical bar ┃ xE2
in the output. That  is,  after the word "Count"  0x436F756E74  I would except just one blank x20, 
then a vertical bar xE2, followed immediatly by a  line feed 0A. 
But instead there are many blanks   between the vertical bar and the end of line at x0A
I've highlighted the example in red.

[Z.txt](https://github.com/user-attachments/files/18201894/Z.txt)
![Output dump](https://github.com/user-attachments/assets/4d0fdc33-2331-4ec2-864e-71f14ca8aa0b)

**Reply by TomJGooding** [reply]
Wasn't expecting a hex dump, I assumed you were talking about visual spaces! :laughing:

**Comment by willmcgugan** [comment]
Remove `justify="left"`

**Reply by scottb5612** [reply]
Thanks.  But why should this make a difference?

**Reply by TomJGooding** [reply]
https://rich.readthedocs.io/en/stable/console.html?highlight=justify#justify-alignment

**Comment by scottb5612** [comment]
It isn't about the appearance, it's about the length of the file, which
grows significantly due to the extra spaces appended to each line.
I have no idea why it is doing it.

**Comment by scottb5612** [comment]
Thanks.
